eth/beacon/helpers.py

```diff
@@ -403,6 +408,77 @@ def get_domain(fork_data: 'ForkData',
     ) * 4294967296 + domain_type
 
 
+@to_tuple
+def get_pubkey_for_indices(validators: Sequence[ValidatorRecord],
+                           indices: Sequence[int]) -> Iterable[int]:
+    for index in indices:
+        yield validators[index].pubkey
+
+
+@to_tuple
+def generate_aggregate_pubkeys(state: BeaconState, vote_data: 'SlashableVoteData') -> Iterable[int]:
+    """
+    Computes the aggregate pubkey we expect based on
+    the proof-of-custody indices found in the ``vote_data``.
+    """
+    proof_of_custody_0_indices = vote_data.aggregate_signature_poc_0_indices
+    proof_of_custody_1_indices = vote_data.aggregate_signature_poc_1_indices
+    all_indices = (proof_of_custody_0_indices, proof_of_custody_1_indices)
+    get_pubkeys = functools.partial(get_pubkey_for_indices, state.validators)
+    return map(bls.aggregate_pubkeys,
+               map(get_pubkeys, all_indices))
+
+
+def build_vote_data_messages(vote_data: 'SlashableVoteData') -> Tuple[bytes, bytes]:
+    """
+    Builds the messages that validators are expected to sign for a ``CasperSlashing`` operation.
+    """
+    # TODO: change to hash_tree_root(vote_data) when we have SSZ tree hashing
+    vote_data_root = vote_data.root
+    return (
+        vote_data_root + (0).to_bytes(1, 'big'),
+        vote_data_root + (1).to_bytes(1, 'big'),
+    )
+
+
+def verify_vote_count(vote_data: 'SlashableVoteData', max_casper_votes: int) -> bool:
+    """
+    Ensures we have no more than ``max_casper_votes`` in the ``vote_data``.
```

Comment:
Nitpick: [We use imperative, present tense to describe APIs](https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/documentation.md#grammar). So:
```suggestion
    Ensure we have no more than ``max_casper_votes`` in the ``vote_data``.
```